### PR TITLE
TMDM-15098 Replace password encryption mechanism

### PR DESCRIPTION
**Pull request by htyin**
https://jira.talendforge.org/browse/TMDM-15098
What is the current behavior? (You should also link to an open issue here)

What is the new behavior?
Update password encryption mechanism and related Junit test

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+		try {
+			Cipher cipher = Cipher.getInstance(this.algorithm);
+			cipher.init(Cipher.ENCRYPT_MODE, this.keySpec);
+			return cipher.doFinal(text.getBytes(StandardCharsets.UTF_8));
+		} catch (Exception e) {
+			return null;
+		}
+	}
+
+	public static void main(String[] args) {
```
Comment (zbcui) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java:138]: does this main method needed?

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
+package org.talend.mdm.commmon.util.core;
```
Comment (hw-zhu) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java:1]: Missing copyright.

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.talend.daikon.crypto.CipherSources;
+import org.talend.daikon.crypto.EncodingUtils;
+import org.talend.daikon.crypto.Encryption;
+import org.talend.daikon.crypto.KeySource;
+import org.talend.daikon.crypto.KeySources;
+
+import com.sun.org.slf4j.internal.LoggerFactory;
```
Comment (hw-zhu) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java:19]: don't use the api,  using log4j2 api for all LOG config
eg:     private static final Logger LOGGER = LogManager.getLogger(xxxxx.class);

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+import com.sun.tools.javac.util.StringUtils;
+
+public class AESEncryption {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(AESEncryption.class);
+
+	public static final String KEYS_FILE = "encryption.keys.file";
```
Comment (hw-zhu) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java:26]: Missing //$NON-NLS-1$

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+					}
+				}
+			}
+			if (keySource == null) {
+				keySource = KeySources.file(AES_KEY);
+			}
+			encryption = new Encryption(keySource, CipherSources.getDefault());
+		} catch (Exception e) {
+			LOGGER.error("Error occurs while initial AESEncryption. ", e);
+		}
```
Comment (hw-zhu) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java:64]: If the block of code just be initialized one time, you'd better using  singleton instance.

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+	private byte[] key;
+
+	private String algorithm;
+
+	public AESEncryption() {
+		try {
+			String keyfile = System.getProperty(KEYS_FILE);
+			KeySource keySource = null;
+			if (StringUtils.isEmpty(keyfile)) {
+				keyfile = System.getProperty("mdm.root") + "/conf/aeskey.dat";
```
Comment (hw-zhu) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java:43]: Between Unit and Windows have different separator, using `System.getProperty("file.separator")` instead

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+	private byte[] key;
+
+	private String algorithm;
+
+	public AESEncryption() {
+		try {
+			String keyfile = System.getProperty(KEYS_FILE);
+			KeySource keySource = null;
+			if (StringUtils.isEmpty(keyfile)) {
```
Comment (hw-zhu) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java:42]: if keyfile is not empty how to deal with it?

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java (hunk 1 above)
Comment (htyin) [reply main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java:138]: Yes, it is for encrypt tool.

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java (hunk 7 above)
Comment (htyin) [reply main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/AESEncryption.java:42]: keySource = KeySources.file(AES_KEY);

**Review by hw-zhu — APPROVED** [review]

**Pull request merged**